packages/allow-scripts/src/setup.js

```diff
@@ -186,13 +164,395 @@ function editPackageJson() {
+  }
+}
+
+/**
+ * @typedef SetDefaultConfigurationOpts
+ * @property {string} rootDir
+ */
+
+/**
+ * @param {SetDefaultConfigurationOpts} param0
+ */
+async function setDefaultConfiguration({ rootDir }) {
+  const conf = await loadAllPackageConfigurations({ rootDir })
+  const {
+    configs: { lifecycle, bin },
+    somePoliciesAreMissing,
+  } = conf
+
+  console.log('\n@lavamoat/allow-scripts automatically updating configuration')
+
+  if (!somePoliciesAreMissing) {
+    console.log('\nconfiguration looks good as is, no changes necessary')
+    return
+  }
+
+  console.log('\nadding configuration:')
+
+  lifecycle.missingPolicies.forEach((pattern) => {
+    console.log(`- lifecycle ${pattern}`)
+    lifecycle.allowConfig[pattern] = false
+  })
+
+  if (FEATURE.bins && bin.somePoliciesAreMissing) {
+    bin.allowConfig = prepareBinScriptsPolicy(bin.binCandidates)
+    console.log(
+      `- bin scripts linked: ${Object.keys(bin.allowConfig).join(',')}`
+    )
+  }
+
+  // update package json
+  await savePackageConfigurations({
+    rootDir,
+    conf,
+  })
+}
+
+// internals
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function addInstallParentDir(filename) {
+  const rootDir = process.env.INIT_CWD || process.cwd()
+  return path.join(rootDir, filename)
+}
+
+/**
+ * @param {string} entry
+ * @param {string} file
+ * @returns {boolean}
+ */
+function isEntryPresent(entry, file) {
+  const rcPath = addInstallParentDir(file)
+  if (!existsSync(rcPath)) {
+    return false
+  }
+  const rcFileContents = readFileSync(rcPath, 'utf8')
+  return rcFileContents.includes(entry)
+}
+
 /**
  * @typedef WriteRcFileContentOpts
  * @property {string} file
  * @property {string} entry
  */
 
 /**
- * @typedef AreBinsBlockedOpts
- * @property {boolean} [noMemoization] Turn off memoization, make a fresh lookup
+ * @param {WriteRcFileContentOpts} param0
  */
+function writeRcFileContent({ file, entry }) {
+  const rcPath = addInstallParentDir(file)
+
+  if (isEntryPresent(entry, file)) {
+    console.log(
+      `@lavamoat/allow-scripts: file ${rcPath} already exists with entry: ${entry}.`
+    )
+  } else {
+    appendFileSync(rcPath, entry + '\n')
+    console.log(`@lavamoat/allow-scripts: added entry to ${rcPath}: ${entry}.`)
+  }
+}
+
+/**
+ * @param {BinCandidates} binCandidates
+ */
+function prepareBinScriptsPolicy(binCandidates) {
+  /** @type {Record<string, string>} */
+  const policy = {}
+  // pick direct dependencies without conflicts and enable them by default unless colliding with bannedBins
+  for (const [bin, infos] of binCandidates.entries()) {
+    const binsFromDirectDependencies = infos.filter((i) => i.isDirect)
+    if (binsFromDirectDependencies.length === 1 && !bannedBins.has(bin)) {
+      // there's no conflicts, seems fairly obvious choice to put it up
+      policy[bin] = binsFromDirectDependencies[0].fullLinkPath
+    }
+  }
+  return policy
+}
+
+/**
+ * @typedef SavePackageConfigurationsOpts
```

Comment:
Note: Previously the inlined typedefs were a bit all over the place. Aside from the `import` of allow-scripts library types in the top, they are now declared just prior to their first usage.